src/main/blackbox/blackbox.c

```diff
@@ -812,11 +812,7 @@ static bool sendFieldDefinition(const char * const *headerNames, unsigned int he
             return false; //Someone probably called us again after we had already completed transmission
         }
 
-        charsWritten = blackboxPrint("H Field ");
-        charsWritten += blackboxPrint(headerNames[xmitState.headerIndex]);
-        blackboxWrite(':');
-        charsWritten++;
-
+        charsWritten = blackboxPrintf("H Field %s:", headerNames[xmitState.headerIndex]);
```

Comment:
printf() buffering would require an actual RAM buffer and copies to be made in and out of it. If the flashfs buffer is looking full, blackboxPrint() allows the string the user passed in (in this case, stored in flash) to written through directly to the dataflash, bypassing any caching in memory so no data needs to be copied.

IMO saving cycles is worthwhile here, even though this is only running at arm time, because of the recent introduction of the code to run the PID loop even at zero throttle. The blackbox header writing happens while the craft is armed.